osquery/tables/system/windows/drivers.cpp

```diff
@@ -173,79 +146,140 @@ Status getDeviceProperty(const device_infoset_t& infoset,
+  GetSystemDirectory(systemRoot, MAX_PATH);
+
+  return systemRoot +
+         boost::regex_replace(path, boost::regex("^.*?system32"), "");
 }
 
 QueryData genDrivers(QueryContext& context) {
   QueryData results;
-
-  auto devInfoset = setupDevInfoSet();
+  auto devInfoset = setupDevInfoSet(DIGCF_ALLCLASSES | DIGCF_PRESENT);
   if (devInfoset == nullptr) {
     win32LogWARNING("Error getting device handle");
     return results;
   }
 
+  std::map<std::string, Row> apiDevices;
   std::vector<SP_DEVINFO_DATA> devices;
   auto ret = getDeviceList(devInfoset, devices);
   if (!ret.ok()) {
     win32LogWARNING(ret.getMessage(), ret.getCode());
     return results;
   }
 
+  // Then, leverage the Windows APIs to get whatever remains
   for (auto& device : devices) {
     char devId[MAX_DEVICE_ID_LEN] = {0};
     if (CM_Get_Device_ID(device.DevInst, devId, MAX_DEVICE_ID_LEN, 0) !=
         CR_SUCCESS) {
       win32LogWARNING("Failed to get device ID");
-      return QueryData();
+      continue;
     }
 
-    SP_DRVINFO_DATA drvInfo = {0};
-    SP_DRVINFO_DETAIL_DATA drvInfoDetail = {0};
-    ret = getDeviceDriverInfo(devInfoset, device, drvInfo, drvInfoDetail);
-
     Row r;
-    r["device_id"] = devId;
-    r["inf"] = drvInfoDetail.InfFileName;
-    r["provider"] = drvInfo.ProviderName;
-    r["manufacturer"] = drvInfo.MfgName;
-    r["date"] = std::to_string(osquery::filetimeToUnixtime(drvInfo.DriverDate));
-    r["description"] = drvInfo.Description;
-    ULARGE_INTEGER version;
-    version.QuadPart = drvInfo.DriverVersion;
-    r["version"] = std::to_string(HIWORD(version.HighPart)) + "." +
-                   std::to_string(HIWORD(version.LowPart)) + "." +
-                   std::to_string(LOWORD(version.HighPart)) + "." +
-                   std::to_string(LOWORD(version.LowPart));
-
     for (const auto& elem : kAdditionalDeviceProps) {
       std::string val;
       ret = getDeviceProperty(devInfoset, device, elem.second, val);
       r[elem.first] = std::move(val);
     }
 
-    if (r.count("driver_key") > 0) {
-      if (!r.at("driver_key").empty()) {
-        r["driver_key"].insert(0, kDriverKeyPath);
-      }
+    if (r.count("driver_key") > 0 && !r.at("driver_key").empty()) {
+      r["driver_key"].insert(0, kDriverKeyPath);
+    }
+
+    if (r.count("service") > 0 && !r.at("service").empty()) {
+      std::string svcKey = kServiceKeyPath + r["service"];
+      r["service_key"] = svcKey;
+      r["image"] = getDriverImagePath(svcKey);
     }
-    if (r.count("service") > 0) {
-      if (!r.at("service").empty()) {
-        r["service_key"] = kServiceKeyPath + r["service"];
-        r["image"] = getDriverImagePath(r["service_key"]);
-      }
+    apiDevices[devId] = r;
+  }
+
+  WmiRequest wmiSignedDriverReq("select * from Win32_PnPSignedDriver");
+  auto& wmiResults = wmiSignedDriverReq.results();
+
+  if (wmiResults.empty()) {
+    LOG(WARNING) << "Failed to query device drivers via WMI";
+    return {};
```

Comment:
Is there really nothing worth returning at this point, even after the above work?